lib/saluki-components/src/sources/checks/scheduler.rs

```diff
@@ -0,0 +1,506 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::{Arc, Mutex, RwLock};
+use std::time::Duration;
+
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+use tokio::time;
+use tracing::{debug, error, info};
+
+use crate::sources::checks::Check;
+
+/// Message type for worker communication
+enum WorkerMessage {
+    RunCheck(Arc<dyn Check + Send + Sync>),
+    Shutdown,
+}
+
+/// A scheduler that manages the execution of checks.
+/// It maintains a dynamic pool of workers and organizes checks by their intervals.
+pub struct Scheduler {
```

Comment:
I think this could be restructured a little differently to end up with a more structured approach to the concurrency involved, but I had a few behavioral questions first:

- is the scheduler meant to actually drop the scheduling of checks if the check runners are busy?
- is running the checks _exactly_ on their interval a requirement or do we only care that we've waited for `interval` time between each run of a check?